torch/testing/_internal/common_fsdp.py

```diff
@@ -1352,6 +1355,36 @@ def _test_fsdp_parity(
             )
 
 
+def test_compiled_fsdp():
+    class FullyShardPatch(Enum):
+        @staticmethod
+        def fully_shard_with_compiled_compute(*args, **kwargs):
+            args[0].forward = torch.compile(args[0].forward)
```

Comment:
wow! just get to know nn.Module.compile. will give it a try

Reply:
updated the PR with ``nn.Module.compile``

Reply:
Another thing to be wary of: https://github.com/pytorch/pytorch/issues/117584 

In practice for transformers, we mainly care about compiling the transformer blocks. The embedding and final projection are less important (since they consist of a smaller portion of the FLOPs). We should be wary of a case where, if we compile the root module and the transformer blocks, we error, but we could have just only compiled the transformer blocks.

For example, for the fp8 work right now, this is the case. We cannot compile the root but can compile the transformer blocks.

Reply:
what do you think of ``@test_compiled_fsdp(compile_on_module=TransformerBlock)`` ? whether or not compiling root module sounds like model specific

Reply:
Specifying which modules to compile seems reasonable to me!

Reply:
updated PR. For transformers, only compile blocks